packages/client/modules/email/components/SummaryEmail/MeetingSummaryEmail/ExportAllTasks.tsx

```diff
@@ -0,0 +1,188 @@
+
+const integrationToServiceName: Record<TaskServiceEnum, string> = {
+  PARABOL: 'Parabol',
+  github: Providers.GITHUB_NAME,
+  jira: 'Jira',
+  jiraServer: Providers.JIRA_SERVER_NAME,
+  azureDevOps: Providers.AZUREDEVOPS_NAME,
+  gitlab: Providers.GITHUB_NAME
+}
+
+interface Props {
+  meetingRef: ExportAllTasks_meeting$key
+}
+
+const ExportAllTasks = (props: Props) => {
+  const {meetingRef} = props
+  const mutationProps = useMutationProps()
+  const atmosphere = useAtmosphere()
+  const {onCompleted, onError, submitMutation, submitting} = mutationProps
+  const {togglePortal, originRef, menuPortal, menuProps, loadingWidth, loadingDelay} = useMenu(
+    MenuPosition.UPPER_RIGHT,
+    {
+      loadingWidth: 200
+    }
+  )
+  const meeting = useFragment(
+    graphql`
+      fragment ExportAllTasks_meeting on NewMeeting {
+        id
+        meetingType
+        teamId
+        ...ExportAllTasksMenuRoot_meeting
+        ... on RetrospectiveMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+        ... on ActionMeeting {
+          tasks {
+            id
+            taskService
+            integration {
+              id
+            }
+          }
+        }
+      }
+    `,
+    meetingRef
+  )
+
+  // :TODO: (jmtaber129): Get this information from the tasks integration.
+  const [pushedIntegrationLabel, setPushedIntegrationLabel] = useState<string | null>(null)
+
+  const {id: meetingId, meetingType, tasks, teamId} = meeting
+  if (!tasks) {
+    return null
+  }
+
+  const filteredTasks = tasks.filter((task) => !task.integration)
+  const taskServices = Array.from(
+    tasks.reduce(
+      (serviceSet, task) => (task.taskService ? serviceSet.add(task.taskService) : serviceSet),
+      new Set<string>()
+    )
+  )
+
+  const handlePushToIntegration = (
+    integrationRepoId: string,
+    integrationProviderService: Exclude<TaskServiceEnum, 'PARABOL'>,
+    integrationLabel?: string
+  ) => {
+    submitMutation()
+    Promise.allSettled(
+      filteredTasks?.map((task) => {
+        return new Promise<void>((resolve, reject) => {
+          const variables = {
+            integrationRepoId,
+            taskId: task.id,
+            integrationProviderService: integrationProviderService
+          }
+          CreateTaskIntegrationMutation(atmosphere, variables, {
+            onError: reject,
+            onCompleted: (res, errors) => {
+              const error = getOnCompletedError(res, errors)
+              if (error) {
+                reject(error)
+              } else {
+                SendClientSegmentEventMutation(atmosphere, 'Bulk Tasks Published', {
+                  teamId,
+                  meetingId,
+                  meetingType,
+                  service: integrationProviderService
+                })
```

Comment:
@tianrunhe these are the metrics changes relevant for your review

Reply:
I don't see much value of having this metric sent from client side. We already have the metric [`Task Published` emitted](https://github.com/ParabolInc/parabol/blob/e24c9bea36fd59920408b043142227b3dc40a6fd/packages/server/utils/analytics/analytics.ts#L352), just not from the `createTaskIntegration` mutation. If it's not too much extra work, would you please add it there instead?

Reply:
We want to specifically track when a bulk export finishes, which is where this is called. Could you elaborate on where you're thinking this should go?

Reply:
Took a second look and test locally. Actually this metric right now is kinda duplicate of what's already been emitted in CreateTaskIntegrationMutation [here](https://github.com/ParabolInc/parabol/blob/e24c9bea36fd59920408b043142227b3dc40a6fd/packages/client/mutations/CreateTaskIntegrationMutation.ts#L252).
I do think there's some value tracking that the button was clicked and user batch published some tasks from this particular location, so maybe after the loop we emit 1 overall metric. The name and attributes can stay the same.

Reply:
Ah yea you're right, I meant for this to be called once at the end but this was just in the wrong place. I moved this to the callback for when all mutations complete, rather than each time a mutation completes.